src/Google/GlobalSiteTag.php

```diff
@@ -155,13 +189,164 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
 		$order->update_meta_data( self::ORDER_CONVERSION_META_KEY, 1 );
 		$order->save_meta_data();
 
-		printf(
-			'<script>gtag("event", "conversion", {"send_to": "%s","value": "%s","currency": "%s","transaction_id": "%s"});</script>',
+		$conversion_gtag_info =
+		sprintf(
+			'gtag("event", "conversion", {
+			send_to: "%s",
+			value: "%s",
+			currency: "%s",
+			transaction_id: "%s"});',
 			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
 			esc_js( $order->get_total() ),
 			esc_js( $order->get_currency() ),
 			esc_js( $order->get_id() ),
 		);
+		wp_print_inline_script_tag( $conversion_gtag_info );
+
+		// Get the item info in the order
+		$item_info = [];
+		foreach ( $order->get_items() as $item_id => $item ) {
+			$product_id   = $item->get_product_id();
+			$product_name = $item->get_name();
+			$quantity     = $item->get_quantity();
+			$price        = $item->get_subtotal();
+			$item_info [] = sprintf(
+				'{
+				id: "gla_%s",
+				price: %s,
+				google_business_vertical: "retail",
+				name:"%s",
+				quantity:"%s",
+				}',
+				esc_js( $product_id ),
+				esc_js( $price ),
+				esc_js( $product_name ),
+				esc_js( $quantity ),
+			);
+		}
+
+		// Check if this is the first time customer
+		$is_new_customer = $this->is_first_time_customer( $order->get_billing_email() );
+
+		// Track the purchase page
+		$language = $this->wp->get_locale();
+		if ( 'en_US' === $language ) {
+			$language = 'English';
+		}
+		$purchase_page_gtag =
+		sprintf(
+			'gtag("event", "purchase", {
+			ecomm_pagetype: "purchase",
+			send_to: "%s",
+			transaction_id: "%s",
+			currency: "%s",
+			country: "%s",
+			value: "%s",
+			new_customer: %s,
+			tax: "%s",
+			shipping: "%s",
+			delivery_postal_code: "%s",
+			aw_feed_country: "%s",
+			aw_feed_language: "%s",
+			items: [%s]});',
+			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
+			esc_js( $order->get_id() ),
+			esc_js( $order->get_currency() ),
+			esc_js( $this->wc->get_base_country() ),
+			esc_js( $order->get_total() ),
+			esc_js( $is_new_customer ),
```

Comment:
For reusability of the helper function I think it was much nicer for it to return a boolean. Having the values returned as strings is how we want to represent it in the snippet but not the result we want to return. I'd suggest to go back to the function `is_first_time_customer` returning a boolean and then replace this line with the output we want:

```suggestion
			$is_new_customer ? 'true' : 'false',
```

Reply:
Thanks for your suggestions. Updated.